akka-stream/src/main/scala-jdk9-only/akka/stream/impl/JavaFlowAndRsConverters.scala

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+
+package akka.stream.impl
+
+import java.util.concurrent.Flow
+
+import akka.annotation.InternalApi
+import org.{ reactivestreams => rs }
+import JavaFlowAndRsConverters.Implicits._
+
+/**
+ * Provides converters between Reactive Streams (reactive-streams.org) and their Java 9+ counter-parts,
+ * defined in `java.util.concurrent.Flow.*`.
+ * 
+ * Please note that either of these types are designed for *inter-op* and usually should not be used directly 
+ * in applications. The intended use case is for shared libraries, like database drivers or similar to provide
+ * the inter-operable types, such that other librarie can co-operate with them directly, if that is your use case
+ * and you're using the j.u.c.Flow types, use the [[akka.stream.scaladsl.JavaFlowSupport]] sources/sinks/flows instead.
+ *
+ * The interfaces as well as semantic contract of these sets of interfaces.
+ * Please see https://github.com/reactive-streams/reactive-streams-jvm to learn more about the specification,
+ * and TCK that is available to verify an implementation is a valid implementation.
+ */
+object JavaFlowAndRsConverters {
```

Comment:
the adapters

Reply:
should have internal API marker since it is in `akka.stream.impl` ?

Reply:
Hmm, good question. I wonder - we could make this actual API for the conversions and move it to akka.stream I guess. The nice thing is that it's asJava/asRs rather than the converters which will eventually arrive in RS, and it'll ship out of the box... 

No strong feelings, shall we make it an API or make it internal? I'll mark it internal for now I think.